worlds/messenger/options.py

```diff
@@ -18,28 +18,77 @@ class Logic(Choice):
     The level of logic to use when determining what locations in your world are accessible.
 
     Normal: can require damage boosts, but otherwise approachable for someone who has beaten the game.
-    Hard: has leashing, normal clips, time warps and turtle boosting in logic.
-    OoB: places everything with the minimum amount of rules possible. Expect to do OoB. Not guaranteed completable.
+    Hard: expects more knowledge and tighter execution. has leashing, normal clips and much tighter d-boosting in logic.
     """
     display_name = "Logic Level"
     option_normal = 0
     option_hard = 1
-    option_oob = 2
+    alias_oob = 1
     alias_challenging = 1
 
 
-class PowerSeals(DefaultOnToggle):
-    """Whether power seal locations should be randomized."""
-    display_name = "Shuffle Seals"
-
-
 class MegaShards(Toggle):
     """Whether mega shards should be item locations."""
     display_name = "Shuffle Mega Time Shards"
 
 
+class LimitedMovement(Toggle):
+    """
+    Removes either rope dart or wingsuit from the itempool. Forces logic to at least hard and accessibility to minimal.
+    """
+    display_name = "Limited Movement"
+
+
+class EarlyMed(Toggle):
+    """Guarantees meditation will be found early"""
+    display_name = "Early Meditation"
+
+
+class AvailablePortals(Range):
+    """Number of portals that are available from the start. Autumn Hills, Howling Grotto, and Glacial Peak are currently always available. If portal outputs are not randomized, Searing Crags will also be available."""
+    display_name = "Available Starting Portals"
+    range_start = 3
+    range_end = 6
+    default = 6
```

Comment:
I have my available portals set to 6 (the default) and have portal shuffle: anywhere set. I start the game and only get 3. The code in `generate_early` around available portals seems fine, so it's either a game mod issue, or the docstring for that option is not clear enough if it's an expected restriction with my options

![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/f38ec3b8-2252-4525-a461-fb454adf1444)

![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/1a2d6262-347c-4fa5-8ff6-0ff137aee8ef)

Reply:
the remaining portals open after you return to the HQ. this is intentional because I had some players enter portals without talking to the prophet, and I can't open all 6 portals in the cutscene.